sinks/cortex/cortex.go

```diff
@@ -59,6 +59,7 @@ type CortexMetricSink struct {
 }
 
 var _ sinks.MetricSink = (*CortexMetricSink)(nil)
+var DoIfNotDoneError = errors.New("context finished before completing metrics flush")
```

Comment:
This is not a 1:1 refactor

> New also records the stack trace at the point it was called

errors.New should not be cached for usages

If error checking is necessary, we should introduce an error type and use `errors.Is`